micronaut-bom 2.0.1 requires a non-existent version of ktor-bom (1.3.2)

As the title describes, micronaut-bom v2.0.1 relies on a non-existent version of ktor-bom. It seems to be a configuration error in pom.xml. micronaut-bom v2.0.0 doesn't have this issue.


ktor-bom 1.3.2 is specified in micronaut-bom's pom. However, the version doesn't exist.

<img width="400" src="https://user-images.githubusercontent.com/19658/90721592-0d39a280-e2f4-11ea-8f17-d04c4753dcc3.png">
<img width="400" src="https://user-images.githubusercontent.com/19658/90721598-10349300-e2f4-11ea-8270-2bce23886ac8.png">

* https://search.maven.org/artifact/io.micronaut/micronaut-bom/2.0.1/pom
* https://search.maven.org/artifact/io.ktor/ktor-bom

### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

I don't think Maven/Gradle version matters but I used Apache Maven 3.6.3 for running this.

1. Add `io.micronaut:micronaut-bom:2.0.1` to dependencies
2. Resolve its dependencies
3. Fails with an errorr message saying `Failure to find io.ktor:ktor-bom:pom:1.3.2`

#### Step1

```xml
<dependency>
    <groupId>io.micronaut</groupId>
    <artifactId>micronaut-bom</artifactId>
    <version>2.0.1</version>
    <type>pom</type>
    <scope>import</scope>
</dependency>
```

#### Step3

```
[INFO] Scanning for projects...
Downloading from central: https://repo.maven.apache.org/maven2/io/micronaut/micronaut-bom/2.0.1/micronaut-bom-2.0.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/io/micronaut/micronaut-bom/2.0.1/micronaut-bom-2.0.1.pom (50 kB at 40 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/io/micronaut/views/micronaut-views-bom/2.0.0/micronaut-views-bom-2.0.0.pom
Downloaded from central: https://repo.maven.apache.org/maven2/io/micronaut/views/micronaut-views-bom/2.0.0/micronaut-views-bom-2.0.0.pom (4.0 kB at 17 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/io/micronaut/groovy/micronaut-groovy-bom/2.0.0/micronaut-groovy-bom-2.0.0.pom
Downloaded from central: https://repo.maven.apache.org/maven2/io/micronaut/groovy/micronaut-groovy-bom/2.0.0/micronaut-groovy-bom-2.0.0.pom (3.3 kB at 13 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/io/micronaut/test/micronaut-test-bom/1.2.2/micronaut-test-bom-1.2.2.pom
Downloaded from central: https://repo.maven.apache.org/maven2/io/micronaut/test/micronaut-test-bom/1.2.2/micronaut-test-bom-1.2.2.pom (3.4 kB at 14 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/io/micronaut/data/micronaut-data-bom/1.1.3/micronaut-data-bom-1.1.3.pom
Downloaded from central: https://repo.maven.apache.org/maven2/io/micronaut/data/micronaut-data-bom/1.1.3/micronaut-data-bom-1.1.3.pom (3.4 kB at 14 kB/s)
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] Non-resolvable import POM: Failure to find io.ktor:ktor-bom:pom:1.3.2 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced @ io.micronaut:micronaut-bom:2.0.1, /Users/ksera/.m2/repository/io/micronaut/micronaut-bom/2.0.1/micronaut-bom-2.0.1.pom, line 1224, column 19
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project com.slack.api:bolt-micronaut:1.1.4-SNAPSHOT (/Users/ksera/github/java-slack-sdk/bolt-micronaut/pom.xml) has 1 error
[ERROR]     Non-resolvable import POM: Failure to find io.ktor:ktor-bom:pom:1.3.2 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced @ io.micronaut:micronaut-bom:2.0.1, /Users/ksera/.m2/repository/io/micronaut/micronaut-bom/2.0.1/micronaut-bom-2.0.1.pom, line 1224, column 19 -> [Help 2]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/UnresolvableModelException
```

### Expected Behaviour

All the dependencies are successfully resolved.

### Actual Behaviour

See the above stdout in Step3.

### Environment Information

- **Operating System**: Any
- **Micronaut Version:** 2.0.1
- **JDK Version:** Any

### Example Application

You can easily reproduce the error just by having the above dependency.

@graemerocher 
This is occurring again with 2.0.3:
https://search.maven.org/artifact/io.micronaut/micronaut-bom/2.0.3/pom

```
INFO] Scanning for projects...
Downloading from central: https://repo.maven.apache.org/maven2/io/micronaut/micronaut-bom/2.0.3/micronaut-bom-2.0.3.pom
Downloaded from central: https://repo.maven.apache.org/maven2/io/micronaut/micronaut-bom/2.0.3/micronaut-bom-2.0.3.pom (50 kB at 41 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/io/ktor/ktor-bom/1.3.2/ktor-bom-1.3.2.pom
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] Non-resolvable import POM: Could not find artifact io.ktor:ktor-bom:pom:1.3.2 in central (https://repo.maven.apache.org/maven2) @ io.micronaut:micronaut-bom:2.0.3, /Users/ksera/.m2/repository/io/micronaut/micronaut-bom/2.0.3/micronaut-bom-2.0.3.pom, line 1240, column 19
 @ 
```

--- a/gradle.properties
+++ b/gradle.properties
@@ -139,7 +139,8 @@ methvinDirectoryWatcherVersion=0.9.6
 jnaVersion=5.3.1
 elasticsearchVersion=7.7.0
 kotlinVersion=1.3.72
-ktorVersion=1.3.2
+# Do not upgrade. See https://github.com/micronaut-projects/micronaut-core/issues/3933
+ktorVersion=1.3.1
 postgresDriverVersion=42.2.14
 mysqlDriverVersion=8.0.21
 mariadbDriverVersion=2.6.1

